pkg/antctl/raw/featuregates/command.go

```diff
@@ -106,14 +90,48 @@ func featureGateRequest(cmd *cobra.Command, mode string) error {
 		}
 	}
 	if len(agentGates) > 0 {
-		fmt.Println(output(agentGates, runtime.ModeAgent))
+		output(agentGates, runtime.ModeAgent, cmd.OutOrStdout())
 	}
 	if len(controllerGates) > 0 {
-		fmt.Println(output(controllerGates, runtime.ModeController))
+		fmt.Println()
```

Comment:
No need to check both since it will only get agent feature gates when we run inside a agent Pod. For other two cases (out-of-cluster or inside of a controller Pod) will get and print both agent and controller feature gates.